src/Raven.Server/Rachis/RachisLogHistory.cs

```diff
@@ -489,5 +498,146 @@ public unsafe bool HasHistoryLog(ClusterOperationContext context, BlittableJsonR
         }
+
+        public static bool TryGetLastDeleteDatabaseCommandWithoutErrors(ServerStore serverStore, string databaseName, out long index)
+        {
+            var commandType = nameof(DeleteDatabaseCommand);
+            index = 0;
+
+            var hasExceptions = false;
+
+            //DeleteDatabaseCommand id looks like that: DeleteDatabaseCommand/EncryptedTest_1/C/sadjghsdkjfng
+
+            using (serverStore.Engine.ContextPool.AllocateOperationContext(out ClusterOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                var table = context.Transaction.InnerTransaction.OpenTable(LogHistoryTable, LogHistorySlice);
+
+                foreach (var result in table.SeekBackwardFrom(LogHistoryTable.Indexes[LogHistoryIndexSlice], Slices.AfterAllKeys))
+                {
+                    var entryHolder = result.Result;
+                    var curIndex = ReadIndex(entryHolder);
+                    var guid = ReadGuid(entryHolder);
+                    var type = ReadType(entryHolder);
+                    var state = ReadState(entryHolder);
+
+                    if (type == commandType && state == HistoryStatus.Committed &&
+                        TryGetNodesAndDatabaseFromDatabaseDeleteCommand(guid, out var databaseFromCmd, out var nodes) &&
+                        databaseFromCmd == databaseName &&
+                        (nodes.Count==0 || nodes.Contains(serverStore.NodeTag)))
+                    {
+                        var exceptionType = ReadExceptionType(entryHolder);
+                        if (string.IsNullOrEmpty(exceptionType)==false)
+                        {
+                            hasExceptions = true;
+                            continue; // has errors after apply
+                        }
+
+                        hasExceptions = false;
+                        index = curIndex;
+                        break;
+                    }
+                }
+            }
+
+            return hasExceptions==false;
+        }
+
+        private static bool TryGetNodesAndDatabaseFromDatabaseDeleteCommand(string guid, out string database, out List<string> nodes)
+        {
+            /*the DeleteDatabaseCommand command id should look like that: "db1/A_B_C/a6m8me9843jt"
+                ($"{databaseName}/{string.Join('_', nodes}")/{newGuid}
+            
+                for example:
+                    TestDB_1/A_C_B/sadjghsdkjfng
+                    TestDB_1/C/sadjghsdkjfng
+             */
+            nodes = null;
+            database = string.Empty;
+
+            var guidParts = guid.Split("/").ToArray();
+            if (guidParts.Length != 3)
+                return false;
+
+            nodes = guidParts[1].Split('_').Where(x => x != string.Empty).ToList();
+            database = guidParts[0];
+
+            return true;
+        }
+
+        public static bool TryGetLastRemoveNodeFromDatabaseCommand(ServerStore serverStore, 
```

Comment:
remove the `static` & the `serverStore`

Reply:
In RachisLogHistory every other method you pass a context, so I did the same here.. (I passed a context instead of serverStore)..